### PR TITLE
Bad data popup

### PR DESCRIPTION
**Pull request by brechmos-stsci**
This will show a popup that says there is bad data if it is an unrecognized IFU datafile, OR if there is a problem with the data file and the data loader can't parse it.

One data file that fails will now popup:
image
rather than show a long traceback on the command line.

Another data file (a MUSE format that isn't recognized) now shows:
image
rather than bringing up cubeviz with no data  in the image viewers.

Addresses #190

### CODE REVIEW

**Comment by brechmos-stsci** [comment]
@astrofrog Is there a better way to do de-register members of the data factory?

**Comment by astrofrog** [comment]
@brechmos-stsci - this isn't ideal as if cubeviz is installed, anyone using glue normally will also not have any other data viewers. I think resetting the data factories should only be done if cubeviz is started from the ``cubeviz`` command.

**Comment by brechmos-stsci** [comment]
@astrofrog ok... so create a function in cubeviz data_factories to remove the defaults and then call that from the cubeviz.py startup function (?).

**Comment by astrofrog** [comment]
Yes I think that makes sense!

**Comment by brechmos-stsci** [comment]
Discussed with @astrofrog. I didn't realize the `setup()` function affects *any* installed glue app and not just in `cubeviz` where it is defined.  So, @astrofrog suggested moving the functionality from `setup()` to `main()` so as not to break other Glue apps.

File: cubeviz/cubeviz.py [hunk 1]
```diff
@@ -41,9 +40,9 @@ def main(argv=sys.argv):
     args = parser.parse_known_args(argv[1:])
 
     # Store the args for each ' --data-configs' found on the commandline
-    global global_data_configuration
-    global_data_configuration['data_configs'] = args[0].data_configs if args[0].data_configs else []
-    global_data_configuration['data_configs_show'] = args[0].data_configs_show
+    data_configuration = {}
+    data_configuration['data_configs'] = args[0].data_configs if args[0].data_configs else []
+    data_configuration['data_configs_show'] = args[0].data_configs_show
```
Comment (astrofrog) [inline cubeviz/cubeviz.py]: Do you actually need the dictionary here, or can you just store in variables called ``data_configs`` and ``data_configs_show``?

**Pull request merged**